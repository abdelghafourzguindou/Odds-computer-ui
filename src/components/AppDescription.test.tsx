import React from "react";
import {render} from "@testing-library/react";
import AppDescriptionComponent from "./AppDescription";

describe("AppDescription Tests", () => {

    test("title of the page", () => {
        const title = render(<AppDescriptionComponent />);
        expect(title).toMatchSnapshot();
    });
})


