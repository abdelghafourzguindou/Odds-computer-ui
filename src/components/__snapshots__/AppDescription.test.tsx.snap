// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppDescription Tests title of the page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-0"
      >
        <div
          class="flex flex-column mb-3"
        >
          <img
            alt="Giskard"
            src="giskard.png"
            width="100"
          />
          <img
            alt="Giskard"
            class="ml-auto"
            src="starwars.png"
            width="100"
          />
        </div>
        <p
          class="MuiTypography-root MuiTypography-body2 MuiTypography-gutterBottom css-ejgmvm-MuiTypography-root"
        >
          The application offering users a way to upload a JSON file containing the data intercepted by the rebels about the plans of the Empire and displaying the odds (as a percentage) that the Millennium Falcon reaches Endor in time and saves the galaxy.
        </p>
        <h6
          class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-gutterBottom css-1oxen2w-MuiTypography-root"
        >
          Example of empire.json file:
        </h6>
        <div
          class="react-json-view"
          style="font-family: monospace; cursor: default; background-color: rgb(39, 40, 34); position: relative;"
        >
          <div
            class="pretty-json-container object-container"
          >
            <div
              class="object-content"
            >
              <div
                class="object-key-val"
              >
                <span>
                  <span
                    style="display: inline-block; cursor: pointer;"
                  >
                    <div
                      class="icon-container"
                      style="display: inline-block; width: 17px;"
                    >
                      <span
                        class="collapsed-icon"
                      >
                        <svg
                          fill="currentColor"
                          style="vertical-align: top; color: rgb(174, 129, 255); height: 1em; width: 1em; padding-left: 2px;"
                          viewBox="0 0 15 15"
                        >
                          <path
                            d="M0 14l6-6-6-6z"
                          />
                        </svg>
                      </span>
                    </div>
                    <span
                      style="display: inline-block; color: rgb(249, 248, 245); letter-spacing: 0.5px; font-style: none; vertical-align: top; opacity: 0.85;"
                    >
                      <span
                        class="object-key"
                      >
                        <span
                          style="vertical-align: top;"
                        >
                          "
                        </span>
                        <span>
                          root
                        </span>
                        <span
                          style="vertical-align: top;"
                        >
                          "
                        </span>
                      </span>
                      <span
                        style="display: inline-block; margin: 0px 5px; color: rgb(249, 248, 245); vertical-align: top;"
                      >
                        :
                      </span>
                    </span>
                    <span
                      style="display: inline-block; cursor: pointer; font-weight: bold; color: rgb(249, 248, 245);"
                    >
                      {
                    </span>
                  </span>
                </span>
                <div
                  class="node-ellipsis"
                  style="display: inline-block; color: rgb(253, 151, 31); font-size: 18px; line-height: 10px; cursor: pointer;"
                >
                  ...
                </div>
                <span
                  class="brace-row"
                >
                  <span
                    style="display: inline-block; cursor: pointer; font-weight: bold; color: rgb(249, 248, 245); padding-left: 0px;"
                  >
                    }
                  </span>
                  <div
                    class="object-meta-data"
                    style="display: inline-block; padding: 0px 0px 0px 10px;"
                  >
                    <span
                      class="object-size"
                      style="color: rgb(165, 159, 133); border-radius: 3px; font-style: italic; margin: 0px 6px 0px 0px; cursor: default;"
                    >
                      2
                       item
                      s
                    </span>
                    <span
                      class="copy-to-clipboard-container"
                      style="vertical-align: top; display: none;"
                      title="Copy to clipboard"
                    >
                      <span
                        style="cursor: pointer; display: inline;"
                      >
                        <span
                          class="copy-icon"
                        >
                          <svg
                            fill="currentColor"
                            preserveAspectRatio="xMidYMid meet"
                            style="vertical-align: top; color: rgb(204, 102, 51); font-size: 15px; margin-right: 3px; height: 1em; width: 1em;"
                            viewBox="0 0 40 40"
                          >
                            <g>
                              <path
                                d="m30 35h-25v-22.5h25v7.5h2.5v-12.5c0-1.4-1.1-2.5-2.5-2.5h-7.5c0-2.8-2.2-5-5-5s-5 2.2-5 5h-7.5c-1.4 0-2.5 1.1-2.5 2.5v27.5c0 1.4 1.1 2.5 2.5 2.5h25c1.4 0 2.5-1.1 2.5-2.5v-5h-2.5v5z m-20-27.5h2.5s2.5-1.1 2.5-2.5 1.1-2.5 2.5-2.5 2.5 1.1 2.5 2.5 1.3 2.5 2.5 2.5h2.5s2.5 1.1 2.5 2.5h-20c0-1.5 1.1-2.5 2.5-2.5z m-2.5 20h5v-2.5h-5v2.5z m17.5-5v-5l-10 7.5 10 7.5v-5h12.5v-5h-12.5z m-17.5 10h7.5v-2.5h-7.5v2.5z m12.5-17.5h-12.5v2.5h12.5v-2.5z m-7.5 5h-5v2.5h5v-2.5z"
                              />
                            </g>
                          </svg>
                        </span>
                      </span>
                    </span>
                  </div>
                </span>
              </div>
            </div>
          </div>
        </div>
        <p
          class="MuiTypography-root MuiTypography-body2 MuiTypography-gutterBottom css-ejgmvm-MuiTypography-root"
        >
          <p>
            * countdown: number of days before the Death Star annihilates Endor
          </p>
          <p>
            * bounty_hunters: list of all locations where Bounty Hunter are scheduled to be present.
          </p>
          <p>
            * planet: Name of the planet. It cannot be null or empty.
          </p>
          <p>
            * day: Day the bounty hunters are on the planet. 0 represents the first day of the mission.
          </p>
          <br />
          <p>
            The web page will display the probability of success as a number ranging from 0 to 100%:
          </p>
        </p>
        <div
          class="css-1aiagyj-MuiStack-root"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardInfo MuiAlert-standard css-1fuz9qr-MuiPaper-root-MuiAlert-root"
            role="alert"
          >
            <div
              class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                data-testid="InfoOutlinedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M11,9H13V7H11M12,20C7.59,20 4,16.41 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20, 12C20,16.41 16.41,20 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10, 10 0 0,0 12,2M11,17H13V11H11V17Z"
                />
              </svg>
            </div>
            <div
              class="MuiAlert-message css-1pxa9xg-MuiAlert-message"
            >
              0% if the Millennium Falcon cannot reach Endor before the Death Star annihilates Endor
            </div>
          </div>
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardInfo MuiAlert-standard css-1fuz9qr-MuiPaper-root-MuiAlert-root"
            role="alert"
          >
            <div
              class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                data-testid="InfoOutlinedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M11,9H13V7H11M12,20C7.59,20 4,16.41 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20, 12C20,16.41 16.41,20 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10, 10 0 0,0 12,2M11,17H13V11H11V17Z"
                />
              </svg>
            </div>
            <div
              class="MuiAlert-message css-1pxa9xg-MuiAlert-message"
            >
              x% if the Millennium Falcon can reach Endor before the Death Star annihilates Endor but might be captured by bounty hunters.
            </div>
          </div>
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardInfo MuiAlert-standard css-1fuz9qr-MuiPaper-root-MuiAlert-root"
            role="alert"
          >
            <div
              class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                data-testid="InfoOutlinedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M11,9H13V7H11M12,20C7.59,20 4,16.41 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20, 12C20,16.41 16.41,20 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10, 10 0 0,0 12,2M11,17H13V11H11V17Z"
                />
              </svg>
            </div>
            <div
              class="MuiAlert-message css-1pxa9xg-MuiAlert-message"
            >
              100% if the Millennium Falcon can reach Endor before the Death Star annihilates Endor without crossing a planet with bounty hunters on it.
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-0"
    >
      <div
        class="flex flex-column mb-3"
      >
        <img
          alt="Giskard"
          src="giskard.png"
          width="100"
        />
        <img
          alt="Giskard"
          class="ml-auto"
          src="starwars.png"
          width="100"
        />
      </div>
      <p
        class="MuiTypography-root MuiTypography-body2 MuiTypography-gutterBottom css-ejgmvm-MuiTypography-root"
      >
        The application offering users a way to upload a JSON file containing the data intercepted by the rebels about the plans of the Empire and displaying the odds (as a percentage) that the Millennium Falcon reaches Endor in time and saves the galaxy.
      </p>
      <h6
        class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-gutterBottom css-1oxen2w-MuiTypography-root"
      >
        Example of empire.json file:
      </h6>
      <div
        class="react-json-view"
        style="font-family: monospace; cursor: default; background-color: rgb(39, 40, 34); position: relative;"
      >
        <div
          class="pretty-json-container object-container"
        >
          <div
            class="object-content"
          >
            <div
              class="object-key-val"
            >
              <span>
                <span
                  style="display: inline-block; cursor: pointer;"
                >
                  <div
                    class="icon-container"
                    style="display: inline-block; width: 17px;"
                  >
                    <span
                      class="collapsed-icon"
                    >
                      <svg
                        fill="currentColor"
                        style="vertical-align: top; color: rgb(174, 129, 255); height: 1em; width: 1em; padding-left: 2px;"
                        viewBox="0 0 15 15"
                      >
                        <path
                          d="M0 14l6-6-6-6z"
                        />
                      </svg>
                    </span>
                  </div>
                  <span
                    style="display: inline-block; color: rgb(249, 248, 245); letter-spacing: 0.5px; font-style: none; vertical-align: top; opacity: 0.85;"
                  >
                    <span
                      class="object-key"
                    >
                      <span
                        style="vertical-align: top;"
                      >
                        "
                      </span>
                      <span>
                        root
                      </span>
                      <span
                        style="vertical-align: top;"
                      >
                        "
                      </span>
                    </span>
                    <span
                      style="display: inline-block; margin: 0px 5px; color: rgb(249, 248, 245); vertical-align: top;"
                    >
                      :
                    </span>
                  </span>
                  <span
                    style="display: inline-block; cursor: pointer; font-weight: bold; color: rgb(249, 248, 245);"
                  >
                    {
                  </span>
                </span>
              </span>
              <div
                class="node-ellipsis"
                style="display: inline-block; color: rgb(253, 151, 31); font-size: 18px; line-height: 10px; cursor: pointer;"
              >
                ...
              </div>
              <span
                class="brace-row"
              >
                <span
                  style="display: inline-block; cursor: pointer; font-weight: bold; color: rgb(249, 248, 245); padding-left: 0px;"
                >
                  }
                </span>
                <div
                  class="object-meta-data"
                  style="display: inline-block; padding: 0px 0px 0px 10px;"
                >
                  <span
                    class="object-size"
                    style="color: rgb(165, 159, 133); border-radius: 3px; font-style: italic; margin: 0px 6px 0px 0px; cursor: default;"
                  >
                    2
                     item
                    s
                  </span>
                  <span
                    class="copy-to-clipboard-container"
                    style="vertical-align: top; display: none;"
                    title="Copy to clipboard"
                  >
                    <span
                      style="cursor: pointer; display: inline;"
                    >
                      <span
                        class="copy-icon"
                      >
                        <svg
                          fill="currentColor"
                          preserveAspectRatio="xMidYMid meet"
                          style="vertical-align: top; color: rgb(204, 102, 51); font-size: 15px; margin-right: 3px; height: 1em; width: 1em;"
                          viewBox="0 0 40 40"
                        >
                          <g>
                            <path
                              d="m30 35h-25v-22.5h25v7.5h2.5v-12.5c0-1.4-1.1-2.5-2.5-2.5h-7.5c0-2.8-2.2-5-5-5s-5 2.2-5 5h-7.5c-1.4 0-2.5 1.1-2.5 2.5v27.5c0 1.4 1.1 2.5 2.5 2.5h25c1.4 0 2.5-1.1 2.5-2.5v-5h-2.5v5z m-20-27.5h2.5s2.5-1.1 2.5-2.5 1.1-2.5 2.5-2.5 2.5 1.1 2.5 2.5 1.3 2.5 2.5 2.5h2.5s2.5 1.1 2.5 2.5h-20c0-1.5 1.1-2.5 2.5-2.5z m-2.5 20h5v-2.5h-5v2.5z m17.5-5v-5l-10 7.5 10 7.5v-5h12.5v-5h-12.5z m-17.5 10h7.5v-2.5h-7.5v2.5z m12.5-17.5h-12.5v2.5h12.5v-2.5z m-7.5 5h-5v2.5h5v-2.5z"
                            />
                          </g>
                        </svg>
                      </span>
                    </span>
                  </span>
                </div>
              </span>
            </div>
          </div>
        </div>
      </div>
      <p
        class="MuiTypography-root MuiTypography-body2 MuiTypography-gutterBottom css-ejgmvm-MuiTypography-root"
      >
        <p>
          * countdown: number of days before the Death Star annihilates Endor
        </p>
        <p>
          * bounty_hunters: list of all locations where Bounty Hunter are scheduled to be present.
        </p>
        <p>
          * planet: Name of the planet. It cannot be null or empty.
        </p>
        <p>
          * day: Day the bounty hunters are on the planet. 0 represents the first day of the mission.
        </p>
        <br />
        <p>
          The web page will display the probability of success as a number ranging from 0 to 100%:
        </p>
      </p>
      <div
        class="css-1aiagyj-MuiStack-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardInfo MuiAlert-standard css-1fuz9qr-MuiPaper-root-MuiAlert-root"
          role="alert"
        >
          <div
            class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
              data-testid="InfoOutlinedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M11,9H13V7H11M12,20C7.59,20 4,16.41 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20, 12C20,16.41 16.41,20 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10, 10 0 0,0 12,2M11,17H13V11H11V17Z"
              />
            </svg>
          </div>
          <div
            class="MuiAlert-message css-1pxa9xg-MuiAlert-message"
          >
            0% if the Millennium Falcon cannot reach Endor before the Death Star annihilates Endor
          </div>
        </div>
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardInfo MuiAlert-standard css-1fuz9qr-MuiPaper-root-MuiAlert-root"
          role="alert"
        >
          <div
            class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
              data-testid="InfoOutlinedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M11,9H13V7H11M12,20C7.59,20 4,16.41 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20, 12C20,16.41 16.41,20 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10, 10 0 0,0 12,2M11,17H13V11H11V17Z"
              />
            </svg>
          </div>
          <div
            class="MuiAlert-message css-1pxa9xg-MuiAlert-message"
          >
            x% if the Millennium Falcon can reach Endor before the Death Star annihilates Endor but might be captured by bounty hunters.
          </div>
        </div>
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardInfo MuiAlert-standard css-1fuz9qr-MuiPaper-root-MuiAlert-root"
          role="alert"
        >
          <div
            class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
              data-testid="InfoOutlinedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M11,9H13V7H11M12,20C7.59,20 4,16.41 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20, 12C20,16.41 16.41,20 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10, 10 0 0,0 12,2M11,17H13V11H11V17Z"
              />
            </svg>
          </div>
          <div
            class="MuiAlert-message css-1pxa9xg-MuiAlert-message"
          >
            100% if the Millennium Falcon can reach Endor before the Death Star annihilates Endor without crossing a planet with bounty hunters on it.
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
